### PR TITLE
New discord bot

### PR DESCRIPTION
**Pull request by aninrusimha**

### CODE REVIEW

File: apps/discordbot/bot.py [hunk 1]
```diff
@@ -37,16 +42,33 @@ async def on_ready(self):
...
+
+
         msg = message.content.lower()
-        if msg == "piss":
-            await message.channel.send("shid")
-        elif msg == "poo" or msg == "poop":
-            await message.channel.send("funny poopies")
-        elif msg == "uh oh":
-            await message.channel.send("stinky")
+        if "hkn" in msg.lower() and "ieee" in msg.lower():
```
Comment (robertquitt) [inline apps/discordbot/bot.py:68]: Did you mean `if "hkn" in msg or "ieee" in msg:`?

File: apps/discordbot/bot.py [hunk 2]
```diff
@@ -55,8 +77,16 @@ async def on_message(self, message):
 
     async def on_member_join(self, member):
         await member.send(
-            "Welcome to the CSUA discord server! Verify your berkeley.edu email to gain access. Please contact a moderator if you have any issues."
+            "Welcome to the CSUA discord server! First, read the rules in #landing-zone. Thumbs up this message if you agree")
+        channel = member.dm_channel
+        def check_thumb(react, user):
+            return user == member and str(react.emoji) == '👍' # thumbs
+        await self.wait_for('reaction_add', check=check_thumb)
```
Comment (robertquitt) [inline apps/discordbot/bot.py:84]: Does this check that they thumbs up that message? Wouldn't it work for any message they thumbs up? I think you can add `message = await member.send(...)`followed by `react.message==message`? Or maybe `react.message.channel == channel`?

File: apps/discordbot/bot.py [hunk 3]
```diff
@@ -37,16 +42,33 @@ async def on_ready(self):
             self.test_channel = get(self.test_guild.channels, id=TEST_CHANNEL_ID)
             self.hoser_role = get(self.csua_guild.roles, id=HOSER_ROLE_ID)
 
+    async def verify_member_email(self, user):
```
Comment (robertquitt) [inline apps/discordbot/bot.py:45]: Where is this being called?

File: apps/discordbot/bot.py (hunk 1 above)
Comment (robertquitt) [reply apps/discordbot/bot.py:68]: Also, nice.

**Review by robertquitt — CHANGES_REQUESTED** [review]
Looks good so far, have you tested it? Also see the comments I left. Thanks!

File: apps/discordbot/bot.py (hunk 3 above)
Comment (robertquitt) [reply apps/discordbot/bot.py:45]: It looks like you aren't calling this anywhere ... ?

File: apps/discordbot/bot.py [hunk 4]
```diff
@@ -37,16 +42,33 @@ async def on_ready(self):
...
 
+    async def verify_member_email(self, user): 
+        channel = user.dm_channel
+        def check_msg(msg, user):
+            return msg.channel == channel
+        got_email = False
+        while not got_email:
+            msg = await self.wait_for('message', check=check_msg)
+            try:
+                validate_email(msg)
```
Comment (robertquitt) [inline apps/discordbot/bot.py:53]: This doesn't check that their email is a berkeley.edu email. Is that intentional?

File: apps/discordbot/bot.py (hunk 1 above)
Comment (aninrusimha) [reply apps/discordbot/bot.py:68]: I don't want it to trigger it accidentally, so only if both clubs are mentioned will it trigger

File: apps/discordbot/bot.py (hunk 2 above)
Comment (aninrusimha) [reply apps/discordbot/bot.py:84]: makes sense, will add

File: apps/discordbot/bot.py (hunk 3 above)
Comment (aninrusimha) [reply apps/discordbot/bot.py:45]: Not sure how to test it properly

File: apps/discordbot/bot.py (hunk 2 above)
Comment (aninrusimha) [reply apps/discordbot/bot.py:89]: fixed

File: apps/discordbot/bot.py (hunk 3 above)
Comment (aninrusimha) [reply apps/discordbot/bot.py:51]: fixed and tested

File: apps/discordbot/bot.py (hunk 4 above)
Comment (aninrusimha) [reply apps/discordbot/bot.py]: added berkeley.edu email check. At the moment, it has to be @ berkeley.edu so prevent people from doing shenanigans with lists or somesuch.

**Comment by aninrusimha** [comment]
Tested it and it generated role correctly.

File: apps/discordbot/bot.py [hunk 5]
```diff
@@ -1,64 +1,87 @@
...
+
     async def on_ready(self):
         print(f"{self.user} has connected to Discord")
         self.is_phillip = self.user.id == CSUA_PHILBOT_CLIENT_ID
         if self.is_phillip:
+            print("Phillip is in the Office")
             self.csua_guild = get(self.guilds, id=CSUA_GUILD_ID)
-            self.test_guild = get(self.guilds, id=TEST_GUILD_ID)
-            self.test_channel = get(self.test_guild.channels, id=TEST_CHANNEL_ID)
+            # self.test_guild = get(self.guilds, id=TEST_GUILD_ID)
```
Comment (robertquitt) [inline apps/discordbot/bot.py:36]: Nit: Delet

File: apps/discordbot/bot.py [hunk 6]
```diff
@@ -1,64 +1,87 @@
...
+            return react.message ==  msg and str(react.emoji) == '👍' # thumbs
+        await self.wait_for('reaction_add', check=check_thumb)
         await member.send(
-            "Welcome to the CSUA discord server! Verify your berkeley.edu email to gain access. Please contact a moderator if you have any issues."
+            "Verify your berkeley.edu email to gain access. First, pleast type your email. Please contact a moderator if you have any issues."
         )
-        url = "https://www.csua.berkeley.edu" + reverse("discord_register")
-        await member.send(url)
+
+        # url = "https://www.csua.berkeley.edu" + reverse("discord_register")
```
Comment (robertquitt) [inline apps/discordbot/bot.py:83]: Nit: delet

File: apps/discordbot/bot.py [hunk 7]
```diff
@@ -1,64 +1,87 @@
...
 
-CSUA_PHILBOT_CLIENT_ID = 737930184837300274
+intents = discord.Intents.all()
+intents.presences = False
 
-# You may have to modify these
-CSUA_GUILD_ID = 368282532757897217
-HOSER_ROLE_ID = 368285558167830529
+TOKEN = config("DISCORD_TOKEN", default="")
+CSUA_GUILD_ID = int(config("TEST_GUILD", default=368282532757897217))
```
Comment (robertquitt) [inline apps/discordbot/bot.py:18]: config("CSUA_GUILD_ID", cast=int)

Comment (robertquitt) [reply apps/discordbot/bot.py:18]: Same for the other 3

File: apps/discordbot/bot.py [hunk 8]
```diff
@@ -1,64 +1,87 @@
-import os
 import logging
-import time
```
Comment (robertquitt) [inline apps/discordbot/bot.py:3]: Also thank you for removing the unneeded imports!

**Comment by aninrusimha** [comment]
Fixed formatting

**Comment by robertquitt** [comment]
Ship it!

**Review by robertquitt — APPROVED** [review]

**Pull request merged**